### ISSUE TITLE
Benchmark between rollup v3 and v4

### ISSUE THREAD

**Issue by luhc228** [rollup/rollup#5182]
The parser from acorn to swc in Rollup v4.  It seems that there is no performance improvement between Rollup v3 and v4.

Benchmark: https://github.com/luhc228/rollup-benchmark-with-v3-and-v4

**Comment by curran** [comment]
I'd be curious to see a size comparison of the assets (JS+WASM) required for the browser build. The WASM asset for v4 is quite large, around 5MB, whereas the totality of the v3 browser build was relatively small as I recall, but I don't have solid numbers at hand.

![image](https://github.com/rollup/rollup/assets/68416/c2c75e8e-b3ff-4346-9002-8a9bf7afafca)

**Comment by lukastaegert** [comment]
The last bit should be somewhat improved in v4.1.4. The benchmarks are also improving slowly as we optimize some parts of the code and collect low-hanging fruit, though the biggest improvements are yet to come.

**Comment by curran** [comment]
Nice! Amazing! Now it's way smaller at

**Comment by curran** [comment]
Amazing! Now it's way smaller at 1.6MB.

![image](https://github.com/rollup/rollup/assets/68416/39954860-bc33-4995-9ade-30cf3075c782)

**Comment by benmccann** [comment]
Is the size likely to decrease further or is that about as small as it can go? I'm wondering if we should upgrade the Svelte REPL to Rollup 4, but it does add 400KB gzip more (https://github.com/sveltejs/sites/pull/546) and so I don't know if we should wait for further improvements or not.

I don't see in the docs or migration guide why wasm is required now. Is it a requirement that could possibly be removed from the browser bundle in order to keep the smaller file size or is it pretty fundamentally tied in?

**Comment by lukastaegert** [comment]
No, it is unlikely to decrease much further, unfortunately. WASM is a requirement because we switched to a Rust-based parser that cannot be transpiled to JavaScript.

**Comment by samal-rasmussen** [comment]
Upgraded the benchmark from 4.0.2 to 4.9.3 and I am seeing a consistent win by v4, but only by a few seconds.
